### PR TITLE
Feature/spring 3 and dependency modules

### PR DESCRIPTION
**Pull request by theo-ritense**

### CODE REVIEW

File: getting-started/modules/core/valtimo-dependencies.md [hunk 1]
```diff
@@ -33,33 +23,36 @@ whether Maven or Gradle is used:
...
 * [Connector](connector.md)
 * [Contract](contract.md)
+* [Core](core.md)
+* [Dashboard](dashboard.md)
+* Data provider
 * [Document](document.md)
+* [Exporter](exporter.md)
 * [Form](form.md)
-* Form flow
+* [Form flow](form-flow.md)
```
Comment (marijnritense) [inline getting-started/modules/core/valtimo-dependencies.md:45]: Can you split off form-flow and form-flow-valtimo into different pages?

File: getting-started/modules/core/valtimo-dependency-versions.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Valtimo dependencies
+
+## Dependencies
+
+The `valtimo-dependency-versions` module specifies dependency versions for Valtimo.
+When this module is used, please specify dependency versions only when needed. 
+
+> NOTE: This dependency will not add any other dependencies to the classpath. It will only define the versions.
```
Comment (marijnritense) [inline getting-started/modules/core/valtimo-dependency-versions.md:8]:
```suggestion
> **_NOTE:_** This dependency does not add any other dependencies to the classpath. It only defines the versions.
```

Partially suggesting this so it is in line with `/introduction/modules/modules.md`

File: getting-started/modules/core/valtimo-dependency-versions.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+> 
+> Please also take a look at [valtimo-dependencies](valtimo-dependencies.md) or [valtimo-gzac-dependencies](../zgw/valtimo-gzac-dependencies.md) when creating an implementation
```
Comment (marijnritense) [inline getting-started/modules/core/valtimo-dependency-versions.md:10]:
```suggestion
> Please also take a look at [valtimo-dependencies](valtimo-dependencies.md) or [valtimo-gzac-dependencies](../zgw/valtimo-gzac-dependencies.md) when creating an implementation.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Migrating to Spring Boot 3
+
+The steps below will describe most relevant changes to Valtimo.
+However, please also check the following guides when running into issues:
+- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)
+- [Preparing for Spring Security 6](https://docs.spring.io/spring-security/reference/5.8/migration/index.html)
+- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html)
+
+As a result of the Spring Boot 3 upgrade, some other frameworks / dependencies needed to be upgraded as well.
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:9]:
```suggestion
As a consequence of the Spring Boot 3 upgrade, some other frameworks / dependencies were also upgraded. 
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html)
+
+As a result of the Spring Boot 3 upgrade, some other frameworks / dependencies needed to be upgraded as well. 
+Each chapter below describes the changes for the relevant framework or dependency.  
+
+## Spring Boot 3 / Spring 6
+
+- `spring.factories` have been [replaced](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#auto-configuration-files) by `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`
+  - Listed configuration classes need to be annotated with `@AutoConfiguration`. More info can be found [here](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.7-Release-Notes#changes-to-auto-configuration)
+- `@Controller` or `@RestController` annotations are [now mandatory](https://github.com/spring-projects/spring-framework/issues/22154)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:16]:
```suggestion
- `@Controller` or `@RestController` annotations [are now mandatory](https://github.com/spring-projects/spring-framework/issues/22154).
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)
+- [Preparing for Spring Security 6](https://docs.spring.io/spring-security/reference/5.8/migration/index.html)
+- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html)
+
+As a result of the Spring Boot 3 upgrade, some other frameworks / dependencies needed to be upgraded as well. 
+Each chapter below describes the changes for the relevant framework or dependency.  
+
+## Spring Boot 3 / Spring 6
+
+- `spring.factories` have been [replaced](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#auto-configuration-files) by `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:14]:
```suggestion
- `spring.factories` have been [replaced](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#auto-configuration-files) by `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  - Listed configuration classes need to be annotated with `@AutoConfiguration`. More info can be found [here](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.7-Release-Notes#changes-to-auto-configuration)
+- `@Controller` or `@RestController` annotations are [now mandatory](https://github.com/spring-projects/spring-framework/issues/22154)
+  - When declaring the controller beans via autoconfiguration, please also annotate the controller with `com.ritense.valtimo.contract.annotation.@SkipComponentScan` to prevent duplicated beans.
+- `javax` packages [changed](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee) to `jakarta`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:18]:
```suggestion
- `javax` packages [changed](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee) to `jakarta`.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+- `@ConstructingBinding` on  no longer supported/needed for `@ConfigurationProperties` classes.
+- Java 8 is no longer supported. Please use Java 17.
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:20]: This seems like an important thing. Can you move this to the top of the list?

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    .antMatchers(GET, "/api/v1/ping").permitAll();
+```
+New:
+```java
+http.authorizeHttpRequests((requests) ->
+    requests.requestMatchers(antMatcher(GET, "/api/v1/ping")).permitAll()
+);
+```
+- `WebSecurityConfigurerAdapter` has been removed.
+  - Removed `CoreHttpSecurityConfigurerAdapter`. This has been replaced by `ValtimoCoreSecurityFactory`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:39]:
```suggestion
  - Removed `CoreHttpSecurityConfigurerAdapter`. This has been replaced by `ValtimoCoreSecurityFactory`.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+- `AuthorizedUrl.acccess(..)` no longer supports expressions as an argument
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:40]:
```suggestion
- `AuthorizedUrl.acccess(..)` no longer supports expressions as an argument.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  - `WhitelistIpRequest` has been replaced by `WhitelistIpRequestMatcher`. An example can be found in `CamundaCockpitHttpSecurityConfigurer`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:41]:
```suggestion
  - `WhitelistIpRequest` has been replaced by `WhitelistIpRequestMatcher`. An example can be found in `CamundaCockpitHttpSecurityConfigurer`.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+      
+- Cockpit is now whitelist-secured again. Support for CIDRs added to support a range of addresses.
+  - A list of hosts can be set at `valtimo.security.whitelist.hosts`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:44]:
```suggestion
  - A list of hosts can be set at `valtimo.security.whitelist.hosts`.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+- Removed `AuthenticationSecurityConfigurer`. No replacement other than implementing a `SecurityFilterChain` directly.
+- OpenAPI endpoint `/v3/api-docs` is now only available to `ROLE_DEVELOPER`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:46]:
```suggestion
- OpenAPI endpoint `/v3/api-docs` is now only available to `ROLE_DEVELOPER`.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+## Hibernate 6
+
+- Overriding entity properties is no longer supported. As a result:
+  - `ProcessLink` implementations can no longer be a `data` class and cannot override the base properties of `ProcessLink`. 
+  Methods previously generated by the `data` type like `equals(other: Any?)`, `hashCode()` and `copy(...)` should be migrated accordingly.
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:52]: Shouldn't this have a dash as well? If not, then it's weird to turn this into a bullet list.

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+- Changes to `application.yaml`:
+  - Removal of `spring.jpa.hibernate.naming.physical-strategy`. No replacement needed.
+  - Removal of `spring.jpa.hibernate.naming.implicit-strategy`. No replacement needed.
+  - Removal of `spring.jpa.hibernate.use-new-id-generator-mappings`. No replacement needed.
+- `AuditRecordRepository.findByEventAndDocumentId(List<Class<? extends AuditEvent>> eventTypes, UUID documentId, Pageable pageable);` changed to `AuditRecordRepository.findByEventAndDocumentId(List<String> eventTypes, UUID documentId, Pageable pageable);`
+
+## Other   
+
+- Camunda History TTL is now required by default. Either:
+  - Set the [camunda:historyTimeToLive](https://docs.camunda.org/manual/7.20/reference/bpmn20/custom-extensions/extension-attributes/#historytimetolive) via the modeler or XML
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:68]:
```suggestion
  - Set the [camunda:historyTimeToLive](https://docs.camunda.org/manual/7.20/reference/bpmn20/custom-extensions/extension-attributes/#historytimetolive) via the modeler or XML.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+  - Removal of `spring.jpa.hibernate.naming.physical-strategy`. No replacement needed.
+  - Removal of `spring.jpa.hibernate.naming.implicit-strategy`. No replacement needed.
+  - Removal of `spring.jpa.hibernate.use-new-id-generator-mappings`. No replacement needed.
+- `AuditRecordRepository.findByEventAndDocumentId(List<Class<? extends AuditEvent>> eventTypes, UUID documentId, Pageable pageable);` changed to `AuditRecordRepository.findByEventAndDocumentId(List<String> eventTypes, UUID documentId, Pageable pageable);`
+
+## Other   
+
+- Camunda History TTL is now required by default. Either:
+  - Set the [camunda:historyTimeToLive](https://docs.camunda.org/manual/7.20/reference/bpmn20/custom-extensions/extension-attributes/#historytimetolive) via the modeler or XML 
+  - Set the application property `camunda.bpm.generic-properties.properties.enforceHistoryTimeToLive` to `false` if you don't want to enforce a TTL
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:69]:
```suggestion
  - Set the application property `camunda.bpm.generic-properties.properties.enforceHistoryTimeToLive` to `false` if you don't want to enforce a TTL.
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+- Mockito 5 vararg matching [has changed](https://github.com/mockito/mockito/releases/tag/v5.0.0)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:70]:
```suggestion
- Mockito 5 vararg matching [has changed](https://github.com/mockito/mockito/releases/tag/v5.0.0).
```

File: release-notes/major12/12.0.0/valtimo-backend-libraries.md [hunk 18]
```diff
@@ -4,10 +4,12 @@
 
 The following features were added:
 
-* **New feature1**
+* **valtimo-dependency-versions module**
+
+  A new `valtimo-dependency-versions` module has been created,
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/valtimo-backend-libraries.md:9]: Although it's kind of obvious what the role is of this module, can you still write that description down here, and provide an possibly obvious example here of why you would want to use this?

File: release-notes/major12/12.0.0/valtimo-backend-libraries.md [hunk 19]
```diff
@@ -4,10 +4,12 @@
 
 The following features were added:
 
-* **New feature1**
+* **valtimo-dependency-versions module**
+
+  A new `valtimo-dependency-versions` module has been created, 
+  which at the same time removes the version control from `valtimo-dependencies` and `valtimo-gzac-depenencies`
+  More information can be found [here](../../../getting-started/modules/core/valtimo-dependency-versions.md)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/valtimo-backend-libraries.md:11]:
```suggestion
  More information can be found [here](../../../getting-started/modules/core/valtimo-dependency-versions.md).
```

File: release-notes/major12/12.0.0/valtimo-backend-libraries.md [hunk 20]
```diff
@@ -4,10 +4,12 @@
 
 The following features were added:
 
-* **New feature1**
+* **valtimo-dependency-versions module**
+
+  A new `valtimo-dependency-versions` module has been created, 
+  which at the same time removes the version control from `valtimo-dependencies` and `valtimo-gzac-depenencies`
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/valtimo-backend-libraries.md:10]:
```suggestion
  which at the same time removes the version control from `valtimo-dependencies` and `valtimo-gzac-depenencies`.
```

File: release-notes/major12/12.0.0/valtimo-backend-libraries.md [hunk 21]
```diff
@@ -31,9 +33,16 @@ The following bugs were fixed:
 
 The following breaking changes were introduced:
 
-* **Breaking change1**
+* **Upgrade to Spring Boot 3**
+
+  Valtimo has been upgraded to Spring Boot 3.
+  This was needed to keep the product up to date with the latest (transitive) dependencies.
+
+  Migration instructions related to this change can be found [here](spring-boot3-migration)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/valtimo-backend-libraries.md:41]:
```suggestion
  Migration instructions related to this change can be found [here](spring-boot3-migration).
```

File: release-notes/major12/12.0.0/valtimo-backend-libraries.md [hunk 22]
```diff
@@ -31,9 +33,16 @@ The following bugs were fixed:
...
 
-* **Breaking change2**
+* **Valtimo dependencies have changed**
+  `valtimo-dependencies` and `valtimo-gzac-depenencies` should no longer be used for `dependencyManagement` or `platform`. 
+  This functionality has been replaced by [valtimo-dependency-versions](../../../getting-started/modules/core/valtimo-dependency-versions.md)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/valtimo-backend-libraries.md:45]:
```suggestion
  This functionality has been replaced by [valtimo-dependency-versions](../../../getting-started/modules/core/valtimo-dependency-versions.md).
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 23]
```diff
@@ -0,0 +1,70 @@
+# Migrating to Spring Boot 3
+
+The steps below will describe most relevant changes to Valtimo.
+However, please also check the following guides when running into issues:
+- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)
+- [Preparing for Spring Security 6](https://docs.spring.io/spring-security/reference/5.8/migration/index.html)
+- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:7]:
```suggestion
- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html).
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 24]
```diff
@@ -0,0 +1,70 @@
+# Migrating to Spring Boot 3
+
+The steps below will describe most relevant changes to Valtimo.
+However, please also check the following guides when running into issues:
+- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)
+- [Preparing for Spring Security 6](https://docs.spring.io/spring-security/reference/5.8/migration/index.html)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:6]:
```suggestion
- [Preparing for Spring Security 6](https://docs.spring.io/spring-security/reference/5.8/migration/index.html).
```

File: release-notes/major12/12.0.0/spring-boot3-migration.md [hunk 25]
```diff
@@ -0,0 +1,70 @@
+# Migrating to Spring Boot 3
+
+The steps below will describe most relevant changes to Valtimo.
+However, please also check the following guides when running into issues:
+- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/spring-boot3-migration.md:5]:
```suggestion
- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide).
```

File: release-notes/major12/12.0.0/migration.md [hunk 26]
```diff
@@ -2,16 +2,18 @@
...
 
-     Description
-  2. **Step2**
+* **Valtimo dependencies**
+
+  Scope: back-end
 
-     Description
+  `valtimo-dependencies` and `valtimo-gzac-depenencies` should no longer be used for `dependencyManagement` or `platform`.
+  This functionality has been replaced by [valtimo-dependency-versions](../../../getting-started/modules/core/valtimo-dependency-versions.md)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/migration.md:16]:
```suggestion
  This functionality has been replaced by [valtimo-dependency-versions](../../../getting-started/modules/core/valtimo-dependency-versions.md).
```

File: release-notes/major12/12.0.0/migration.md [hunk 27]
```diff
@@ -2,16 +2,18 @@
...
 This page describes how to update Valtimo from the previous version to the current.
 
-* **Breaking change 1/Deprecation 1**
+* **Spring boot 3 upgrade**
 
-  Scope: back-end/front-end
+  Scope: back-end
 
-  1. **Step1**
+  Full migration instructions for this change can be found [here](spring-boot3-migration.md)
```
Comment (marijnritense) [inline release-notes/major12/12.0.0/migration.md:9]:
```suggestion
  Full migration instructions for this change can be found [here](spring-boot3-migration.md).
```

File: SUMMARY.md [hunk 28]
```diff
@@ -199,6 +200,7 @@
       * [Valtimo backend libraries](release-notes/major12/12.0.0/valtimo-backend-libraries.md)
       * [Valtimo frontend libraries](release-notes/major12/12.0.0/valtimo-frontend-libraries.md)
       * [Migration](release-notes/major12/12.0.0/migration.md)
+        * [Spring Boot 3](release-notes/major12/12.0.0/spring-boot3-migration.md)
   * 11.x.x
```
Comment (marijnritense) [inline SUMMARY.md:204]: Can you add the 12.x.x release notes here as well

File: getting-started/modules/core/valtimo-dependencies.md (hunk 1 above)
Comment (theo-ritense) [reply getting-started/modules/core/valtimo-dependencies.md:45]: done

File: release-notes/major12/12.0.0/spring-boot3-migration.md (hunk 8 above)
Comment (theo-ritense) [reply release-notes/major12/12.0.0/spring-boot3-migration.md]: done

File: release-notes/major12/12.0.0/spring-boot3-migration.md (hunk 14 above)
Comment (theo-ritense) [reply release-notes/major12/12.0.0/spring-boot3-migration.md:54]: removed the sub-list for this change

File: release-notes/major12/12.0.0/valtimo-backend-libraries.md (hunk 18 above)
Comment (theo-ritense) [reply release-notes/major12/12.0.0/valtimo-backend-libraries.md:9]: done

**Review by mritense — APPROVED** [review]

File: SUMMARY.md (hunk 28 above)
Comment (marijnritense) [reply SUMMARY.md:205]: Huh, I missed that

**Pull request merged**